server/log.js

```diff
@@ -18,17 +18,40 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-var winston = require('winston');
-require('winston-daily-rotate-file');
-
-/* Initialize rotating logging. */
-exports.init = function(callback) {
-    winston.add(winston.transports.DailyRotateFile, {
-        filename: 'log/quizzard',
-        datePattern: 'yyyy-MM-dd-',
-        prepend: true
-    });
-    callback();
+var fs = require('fs');
+var common = require('./common.js');
+var dateStamp;
+var logger;
+
+// print a normal message
+exports.log = function (text) {
+    init();
+
+    var txt = common.formatString('[{0}] log: {1}', [common.getDate(), text]);
+    console.log(txt);
+}
+
+// print an error message
+exports.error = function (text) {
+    init();
+
+    var txt = common.formatString('[{0}] error: {1}', [common.getDate(), text]);
+    console.error(txt);
 }
 
-exports.logger = winston;
+// initiate the logging with file
+var init = function () {
+    var currentDate = common.getDateByFormat('YYYY-MM-DD');
+    if (!process.env.DEBUG && dateStamp !== currentDate) {
+        dateStamp = currentDate;
+        logger = fs.createWriteStream(__dirname + '/../logs/'+dateStamp+'.log', {'flags':'a'});
+        process.stdout.write = process.stderr.write = logger.write.bind(logger);
+        process.on(
+            'uncaughtException',
+            function(err)
+            {
```

Comment:
can we use the standard way we di bracketing here:
``` javascript
function(err) {
    //contents
}
```

Reply:
yeah my bad. will fix